lib/SILOptimizer/Utils/SILInliner.cpp

```diff
@@ -601,9 +601,17 @@ SILValue SILInlineCloner::borrowFunctionArgument(SILValue callArg,
                                     ->getModule()
                                     .getASTContext()
                                     .LangOpts.EnableExperimentalLexicalLifetimes;
-  if (!AI.getFunction()->hasOwnership() ||
-      (callArg.getOwnershipKind() != OwnershipKind::Owned &&
-       !enableLexicalLifetimes)) {
+  auto argOwnershipRequiresBorrow = [&]() {
+    auto kind = callArg.getOwnershipKind();
+    if (enableLexicalLifetimes) {
+      // Add a lexical borrow scope for guaranteed arguments to ensure that they
+      // continue to remain alive for the duration of the callee even after the
+      // callee has been inlined.
+      return kind == OwnershipKind::Owned || kind == OwnershipKind::Guaranteed;
```

Comment:
This comment is misleading because it seems to be referring to the following condition when refers to ownership. But the ownership kind here is not the argument ownership that it's referring to. It's up to the caller to only call this routine for guaranteed arguments (which has nothing do to with the code here).

This condition seems to be about additionally *avoiding* a borrow scope if the caller's incoming value is trivial, which should be
`return kind != OwnershipKind::None`

Reply:
Fixed!